### PR TITLE
Fix pylint 2.16 warnings

### PR DESCRIPTION
**Pull request by r0qs**
Fixes pylint warnings: https://app.circleci.com/pipelines/github/ethereum/solidity/28336/workflows/ee0dd805-5e24-4b17-9a2f-a37ce00ce7c8/jobs/1257578/parallel-runs/0/steps/0-105

### CODE REVIEW

**Review by matheusaaguiar — DISMISSED** [review]

File: test/formal/checked_int_mul_12.py [hunk 1]
```diff
@@ -39,8 +39,8 @@
 			sol_overflow_check_2 = AND(SLT(X, 0), EQ(Y, min_value))
 			sol_overflow_check = Or(sol_overflow_check_1 != 0, sol_overflow_check_2 != 0)
 		else:
-			sol_overflow_check = (sol_overflow_check_1 != 0)
+			sol_overflow_check = sol_overflow_check_1 != 0
```
Comment (cameel) [inline test/formal/checked_int_mul_12.py]: What do you think about disabling this rule from our `pylintrc` instead? We're doing it pretty much only in places where we have comparisons and I think parentheses add readability in those cases. I'd rather keep this code as is.

File: scripts/bytecodecompare/prepare_report.py [hunk 2]
```diff
@@ -251,7 +251,7 @@ def detect_metadata_cli_option_support(compiler_path: Path):
             f"Compiler output:\n{process.stderr}\n",
             file=sys.stderr
         )
-        raise Exception("Failed to determine if the compiler supports the --metadata option.")
+        raise RuntimeError("Failed to determine if the compiler supports the --metadata option.")
```
Comment (cameel) [inline scripts/bytecodecompare/prepare_report.py:254]: Not sure it's much better. It should have a proper exception type.

I guess we can leave it as is for now though.

File: test/formal/checked_int_mul_12.py (hunk 1 above)
Comment (r0qs) [reply test/formal/checked_int_mul_12.py]: Yeah, both ways are fine for me. When you only have one condition it is not that difficult to read.

File: scripts/bytecodecompare/prepare_report.py (hunk 2 above)
Comment (r0qs) [reply scripts/bytecodecompare/prepare_report.py:254]: For sure. I just added as `RuntimeError` to silent the pylint. We should create proper exceptions for each case indeed.

File: test/formal/checked_int_mul_12.py (hunk 1 above)
Comment (cameel) [reply test/formal/checked_int_mul_12.py]: It always throws me off when you have `=` and `==` in it so I always put a parenthesis on it. And I've seen others do it too. I actually think that conditions should always be parenthesized when you have `=` in there.

Comment (r0qs) [reply test/formal/checked_int_mul_12.py]: haha, no problem. Make sense, let's disable it in our pylintrc then :)

**Comment by matheusaaguiar** [comment]
I have disabled the `C0325: Superfluous-parens` error message and reverted the removal of affected parentheses.

**Review by cameel — APPROVED** [review]

**Comment by cameel** [comment]
Pylint check passed so I'm going to force merge it to speed things but. We'll have a full CI run in the changelog PR anyway.

**Pull request merged**